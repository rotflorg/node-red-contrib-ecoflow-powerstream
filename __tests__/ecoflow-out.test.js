var helper = require('node-red-node-test-helper');
var testNode = require('../dist/ecoflow-out.js');

helper.init(require.resolve('node-red'), {
  functionGlobalContext: { os:require('os') }
});


describe('ecoflow-out Node', function () {
  let dateNowSpy;
  beforeEach(function (done) {
    helper.startServer(done);
    dateNowSpy = jest.spyOn(Date, 'now').mockImplementation(() => 1704063600000);
  })
  afterEach(function (done) {
    dateNowSpy.mockRestore();
    helper.unload();
    helper.stopServer(done);
  });

  it('should be loaded', function (done) {
    const flow = [{ id: 'n1', type: 'ecoflow-out', name: 'test name' }];
    helper.load(testNode, flow, {}, function () {
      const n1 = helper.getNode('n1');
      expect(n1.name).toEqual('test name');
      done();
    });
  });

  it('should convert PowerStream getLatestQuotas payload', function (done) {
    const flow = [{ id: 'n1', type: 'ecoflow-out', name: 'test name', devicetype: 'powerstream', devicesn: 'HW51ZEH000000000', wires: [['n2']] }, { id: 'n2', type: 'helper' }];
    helper.load(testNode, flow, function () {
      const n2 = helper.getNode('n2');
      const n1 = helper.getNode('n1');
      const receivedMessages = [];
      n2.on('input', function(msg) {
        const copy = { ...msg };
        delete copy._msgid;
        receivedMessages.push(copy);
      });
      n1.receive({ payload: { getLatestQuotas: 1 } });
      setTimeout(function() {
        expect(receivedMessages).toMatchSnapshot();
        done();
      }, 1000);
    });
  });

  it('should convert PowerStream setAcWatts payload', function (done) {
    const flow = [{ id: 'n1', type: 'ecoflow-out', name: 'test name', devicetype: 'powerstream', devicesn: 'HW51ZEH000000000', wires: [['n2']] }, { id: 'n2', type: 'helper' }];
    helper.load(testNode, flow, function () {
      const n2 = helper.getNode('n2');
      const n1 = helper.getNode('n1');
      const receivedMessages = [];
      n2.on('input', function(msg) {
        const copy = { ...msg };
        delete copy._msgid;
        receivedMessages.push(copy);
      });
      n1.receive({ payload: { setAcWatts: 100 } });
      setTimeout(function() {
        expect(receivedMessages).toMatchSnapshot();
        done();
      }, 1000);
    });
  });

});
