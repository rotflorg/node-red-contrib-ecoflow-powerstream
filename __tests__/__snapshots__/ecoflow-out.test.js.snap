// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ecoflow-out Node should convert PowerStream getLatestQuotas payload 1`] = `
[
  {
    "payload": {
      "data": [
        10,
        16,
        16,
        32,
        24,
        32,
        112,
        240,
        228,
        199,
        172,
        6,
        186,
        1,
        3,
        105,
        111,
        115,
      ],
      "type": "Buffer",
    },
    "topic": "/app/device/HW51ZEH000000000/thing/property/get",
  },
]
`;

exports[`ecoflow-out Node should convert PowerStream setAcWatts payload 1`] = `
[
  {
    "payload": {
      "data": [
        10,
        61,
        10,
        3,
        8,
        232,
        7,
        16,
        32,
        24,
        53,
        32,
        1,
        40,
        1,
        56,
        3,
        64,
        20,
        72,
        129,
        1,
        80,
        3,
        88,
        1,
        112,
        240,
        228,
        199,
        172,
        6,
        128,
        1,
        19,
        136,
        1,
        1,
        186,
        1,
        3,
        105,
        111,
        115,
        202,
        1,
        16,
        72,
        87,
        53,
        49,
        90,
        69,
        72,
        48,
        48,
        48,
        48,
        48,
        48,
        48,
        48,
        48,
      ],
      "type": "Buffer",
    },
    "topic": "/app/device/HW51ZEH000000000/thing/property/set",
  },
]
`;
